### ISSUE TITLE
libSass 3.3 compatibility

### ISSUE THREAD

**Issue by craigbarnes** [craigbarnes/lua-sass#5]
As [reported](https://github.com/craigbarnes/lua-sass/issues/4#issuecomment-164748566) by @Eroica in #4, libsass [changed](https://github.com/sass/libsass/commit/36feef0c4ccfb77438cba99fcc33094444363054) their headers and broke compatibility again.

Unfortunately the suggested fix:

> ... sass_context.h should be changed to sass/context.h

isn't sufficient on it's own. With this change the module compiles, but fails with the error:

    *** Error in `lua': munmap_chunk(): invalid pointer: 0x0000000000ac4e08 ***
    Aborted

It seems like the internals have also changed somehow.

**Comment by craigbarnes** [comment]
It seems like libsass mutates the input buffer now. I always [assumed](https://github.com/craigbarnes/lua-sass/blob/fe28c4c67c7c0e1b3c794b63d1a0ff66314f3e9e/sass.c#L34-L35) it wasn't marked `const` simply due to sloppy coding, since mutating the input seems a strange thing to do. The Lua input string will just have to be copied before passing to libsass now.

**Comment by craigbarnes** [comment]
Ok, after copying the string, libsass is now complaining that [line 23](https://github.com/craigbarnes/lua-sass/blob/317b506a16c1fde93c8a154c32641819039df4ee/test.lua#L23) in `test.lua` is invalid UTF-8. I'm fairly sure it isn't...

**Comment by craigbarnes** [comment]
According to `iconv` (which I trust much more than `libass`), `test.lua` is all valid UTF-8:

    iconv -f UTF-8 test.lua -o /dev/null

So this seems like a bug in `libass` itself.

**Comment by craigbarnes** [comment]
Turns out the API has changed too. To be honest, I'm thinking of abandoning maintenance of lua-sass. I no longer have any personal use for it and libsass is possibly the worst cowboy coding I've ever seen.

**Comment by Eroica** [comment]
Hi, I also noticed (after I already replied to the other issue) that compatibility is still broken (I just assumed everything was alright after it compiled succesfully, sorry about that). No worries about your feelings towards libsass' and this project's current situation--I'll see whether I'll be able to fix this for my project or perhaps I can build a libsass wrapper based on your code. Still, thank you for taking your time to investigate this issue and your original effort!
